verification/curator-service/api/src/controllers/cases.ts

```diff
@@ -107,6 +107,73 @@ export default class CasesController {
         }
     };
 
+    /**
+     * Validates and upserts the provided cases.
+     *
+     * Executes three operations on the supplied data:
+     *
+     *   1. Geocodes provided cases as required. If there are any issues
+     *      geocoding (which is done serially -- though we should consider
+     *      taking advantage of a batch geocode API), return the results now
+     *      without proceeding to the validation stage.
+     *   2. Performs validation of all provided cases via the data service
```

Comment:
shouldn't we assume geocode is going to be valid and validate the rest before geocoding? validation is free, geocoding is a paid and much slower service.

Reply:
Considered this for the reasons you listed. Also considered failing the geocoding stage fast (stopping on the first error, rather than doing all cases) for similar reasons.

I'm sort of on the fence for this and could go either way. I think our caching is likely to have a high hit rate for bulk geocoding, which abates the biggest downsides. The downside to validating first is that it requires setting dummy location values, and so makes this marginally more complex.

WDYT?

Reply:
I also hope the LRU cache does its job well enough so it's probably not worth the added complexity of dummy location values, fine as is.